src/solc_0.8/defi/rules/ContributionRules.sol

```diff
@@ -0,0 +1,205 @@
+
+        _listERC721[multContract].multipliers = multipliers;
+        _listERC721[multContract].balanceOf = balanceOf;
+        _listERC721[multContract].ids = ids;
+
+        // if it's a new member create a new registry, instead, only update
+        if (_islistERC721Member(multContract) == false) {
+            _listERC721Index.push(multContract);
+            _listERC721[multContract].index = _listERC721Index.length - 1;
+        }
+
+        ERC721MultiplierAdded(contractERC721, multipliers, ids, balanceOf);
+    }
+
+    function getMultiplierERC721List(address reqContract)
+        external
+        view
+        _isContract(reqContract)
+        returns (MultiplierRule memory)
+    {
+        require(_islistERC721Member(IERC721(reqContract)), "ContributionRules: contract is not in the list");
+        return _listERC721[IERC721(reqContract)];
+    }
+
+    function getMultiplierERC1155List(address reqContract)
+        external
+        view
+        _isContract(reqContract)
+        returns (MultiplierRule memory)
+    {
+        require(_islistERC1155Member(IERC1155(reqContract)), "ContributionRules: contract is not in the list");
+        return _listERC1155[IERC1155(reqContract)];
+    }
+
+    function deleteMultiplierERC721fromList(IERC721 reqContract) external _isContract(address(reqContract)) onlyOwner {
+        require(_islistERC721Member(reqContract), "ContributionRules: contract is not in the list");
+        uint256 indexToDelete = _listERC721[reqContract].index;
+        IERC721 addrToMove = _listERC721Index[_listERC721Index.length - 1];
+        _listERC721Index[indexToDelete] = addrToMove;
+        _listERC721[addrToMove].index = indexToDelete;
+        _listERC721Index.pop();
+
+        emit ERC721MultiplierDeleted(address(reqContract));
+    }
+
+    function deleteMultiplierERC1155fromList(IERC1155 reqContract)
+        external
+        _isContract(address(reqContract))
+        onlyOwner
+    {
+        require(_islistERC1155Member(reqContract), "ContributionRules: contract is not in the list");
+        uint256 indexToDelete = _listERC1155[reqContract].index;
+        IERC1155 addrToMove = _listERC1155Index[_listERC1155Index.length - 1];
+        _listERC1155Index[indexToDelete] = addrToMove;
+        _listERC1155[addrToMove].index = indexToDelete;
+        _listERC1155Index.pop();
+
+        emit ERC1155MultiplierDeleted(address(reqContract));
+    }
+
+    function _islistERC721Member(IERC721 reqContract) internal view returns (bool) {
+        if (_listERC721Index.length == 0) return false;
+
+        return (_listERC721Index[_listERC721[reqContract].index] == reqContract);
+    }
+
+    function _islistERC1155Member(IERC1155 reqContract) internal view returns (bool) {
+        if (_listERC1155Index.length == 0) return false;
+
+        return (_listERC1155Index[_listERC1155[reqContract].index] == reqContract);
+    }
+
+    function _multiplierBalanceOfERC721(address account) internal view returns (uint256) {
```

Comment:
why not public ? It could help us debug

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/commit/3eebb77ee852e2dd4fb5f5a788b3ed2ac64fb5bc